tests/unit/test_dataset_builders/test_variables_metadata_with_library_metadata_dataset_builder.py

```diff
@@ -0,0 +1,128 @@
+from unittest.mock import MagicMock, patch
+from cdisc_rules_engine.services.cache.in_memory_cache_service import (
+    InMemoryCacheService,
+)
+from cdisc_rules_engine.utilities.utils import get_standard_details_cache_key
+import pandas as pd
+from cdisc_rules_engine.dataset_builders.variables_metadata_with_library_metadata import (  # noqa: E501
+    VariablesMetadataWithLibraryMetadataDatasetBuilder,
+)
+from cdisc_rules_engine.services.data_services import LocalDataService
+
+
+@patch(
+    "cdisc_rules_engine.services.data_services.LocalDataService.get_dataset",
+)
+@patch(
+    "cdisc_rules_engine.services.data_services.LocalDataService.get_variables_metadata",
+)
+def test_variable_metadata_with_library_metadata_dataset_builder(
+    mock_get_variables_metadata: MagicMock, mock_get_dataset: MagicMock
+):
+    mock_get_variables_metadata.return_value = pd.DataFrame.from_dict(
```

Comment:
Can you add tests for variables that are in the data, but not in the ig or model.
And if you make the other update, add tests for variables that are in the data and model, but not ig.